test/one/profiler/test/convert/OutputConverter.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.profiler.test.convert;
+
+import one.convert.Arguments;
+
+import java.io.IOException;
+
+/*
+Interface to be implemented by various profile output converters in tests.
+ */
+public interface OutputConverter<I, O> {
+
+    O convert(I input, Arguments args) throws IOException;
```

Comment:
3 new classes added to support one conversion function. Let's not overcomplicate it with an architecture for arbitrary conversions: most likely, there will be just two conversions: flamegraph->collapsed and jfr->collapsed. I'd just add two corresponding methods in `Output` class.

Reply:
Created thinking of future unknown conversions. Simplified as you mentioned of conversion of profile output being limited to 2 scenarios.